### PR TITLE
URL generation rule explained for file uploads

### PR DESCRIPTION
**Pull request by sourya**

### CODE REVIEW

File: docs/patterns/fileuploads.rst~ [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+should already know.
+
+Now there are better solutions to that work faster and more reliable.  The
+web changed a lot lately and you can use HTML5, Java, Silverlight or Flash
+to get a nicer uploading experience on the client side.  Look at the
+following libraries for some nice examples how to do that:
+
+-   `Plupload <http://www.plupload.com/>`_ - HTML5, Java, Flash
+-   `SWFUpload <http://www.swfupload.org/>`_ - Flash
+-   `JumpLoader <http://jumploader.com/>`_ - Java
```
Comment (ThiefMaster) [inline docs/patterns/fileuploads.rst~]: I think it's a very bad idea to suggest anything Flash-based here. Much worse, anything Java-based.
Some beginners will think that using Java for this is acceptable. It's not. And Flash is "meh", but not really necessary nowadays.

**Comment by untitaker** [comment]
Why close this?

**Comment by sourya** [comment]
Changing the File upload progress bar part. Will create another pull request soon. :)

**Comment by keyanp** [comment]
@untitaker This looks good to me, could merge it and close #1256?

**Comment by untitaker** [comment]
Ah yes... JS libraries come and go, I'd like to avoid linking to a specific one.

**Comment by keyanp** [comment]
Hmm, what is the quickest way to fix this? Could you merge it and then I just remove the link to dropzonejs? I'm guessing jQuery is here to stay and is safe to leave in.

**Pull request merged**